src/topology/sheaves/sheaf_condition/basis_le.lean

```diff
@@ -0,0 +1,273 @@
+/-
+Copyright (c) 2021 Junyan Xu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Junyan Xu
+-/
+
+import topology.sheaves.sheaf_condition.opens_le_cover
+
+/-!
+# Consequences of the sheaf condition about covers by open sets in a topological basis
+
+Let `B` be a basis of a topological space `X`.
+
+This file shows that if `G` is a sheaf on `X`,
+then `G(U)` is the limit of all `G(V)` with `V ∈ B, V ⊆ U`,
+and if `α : F ⟶ G` is a homomorphism of presheaves on `X`
+defined only on open sets in `B`, it uniquely extends to an
+actual homomorphism on all open sets.
+
+-/
+
+universes v u
+
+noncomputable theory
+
+open category_theory
+open category_theory.limits
+open topological_space
+open opposite
+open topological_space.opens
+
+namespace Top
+
+variables {C : Type u} [category.{v} C] (X : Top.{v})
+
+/-- Structure for indexing a family of open sets in a topological space. -/
+structure opens_index_struct :=
+(ι : Type v)
+(f : ι → opens X)
+-- if ι and B are separate then Lean cannot infer induced category instance on ι
+
+instance : inhabited (opens_index_struct X) := ⟨⟨ulift empty, empty.elim ∘ ulift.down⟩⟩
+
+variables {X} (B : X.opens_index_struct) (U : opens X) (F G : presheaf C X)
+
+/-- The proposition that the range of the family `B` is a basis. -/
+def is_basis_range := is_basis (set.range (B.f))
+
+namespace presheaf
+
+namespace sheaf_condition
+
+/--
+The subtype of indices `i` of the family `B` (normally an indexed basis) whose
+corresponding open sets `B.f i` are contained in `U`.
+-/
+@[nolint has_inhabited_instance]
+def basis_le : Type v := { i // B.f i ≤ U }
```

Comment:
Why make this definition when you can just write `category_theory.full_subcategory B`?

Reply:
I blindly copied from opens_le_cover.lean and didn’t give it a second thought … I’ll check to see if this definition has any advantages.